### ISSUE TITLE
Cannot read property 'resolve_dependency_path' of undefined

### ISSUE THREAD

**Issue by gibbymt** [ConsenSys/truffle#362]
- [x ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this 

---------------------------

## Issue

When I do a fresh init and compile I get this error. Others in the glitter chat are seeing this issue as well. Downgrading to 3.1.9 solves the issue.

## Steps to Reproduce

- npm npm install -g truffle
- truffle init webpack
- truffle compile

## Expected Behavior

compile should succeed

## Actual Results

```
truffle compile
/usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:204
            dependency_path = source.resolve_dependency_path(import_path, dependency_path);
                                    ^

TypeError: Cannot read property 'resolve_dependency_path' of undefined
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:204:37
    at Array.map (native)
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:201:27
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/index.js:77:5
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/node_modules/async/internal/onlyOnce.js:12:16
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/node_modules/async/whilst.js:63:18
    at apply (/usr/local/lib/node_modules/truffle/node_modules/lodash/_apply.js:15:25)
    at /usr/local/lib/node_modules/truffle/node_modules/lodash/_overRest.js:32:12
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/index.js:62:7
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/fs.js:41:12
    at tryToString (fs.js:455:3)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:442:12)
```

## Environment

* Operating System: Mac 10.12.3
* Truffle version: 3.11
* Ethereum client: 
* node version: 6.9.3
* npm version: 4.1.1

**Comment by lizardkinglin** [comment]
Dear gibbymt,

I suffer the the same issue with you. Bad engineer luck also. ^^ Could you show me to downgrade the truffle version ? Thanks a lot for your help.

**Comment by gibbymt** [comment]
npm install -g truffle@3.1.9 

npm is the package manager, install is the action, and '-g' means you want this to be a global change and '@3.1.9' is the version

**Comment by hitchcott** [comment]
It's caused because `truffle@3.1.1` is installing `truffle-compile@1.1.0` (which was recently published for the `3.1.9` release) instead of `truffle-compile@1.0.0`, npm is resolving to `1.1.0` because it's a minor upgrade.

Few fix suggestions:

*`truffle-compile@1.1.0` is unpublished and then republished with a new major release `2.0.0`
* Publish a new backwards-compatible `truffle-compile@1.1.1` 
* The non-beta truffle is re-published with `"truffle-compile": "1.0.0"` instead of `"truffle-compile": "^1.0.0"`

**Comment by tcoulter** [comment]
This was due to improper dependency management (my fault). This has been fixed in 3.1.2. 

Note that 3.1.9 is a beta release, preparing for 3.2.0
